File: solutions/swift/collatz-conjecture/1/Sources/CollatzConjecture/CollatzConjecture.swift
enum CollatzConjectureError: Error { case negative }

class CollatzConjecture {
  static func steps(_ number: Int) throws -> Int? {
    guard number > 0 else { throw CollatzConjectureError.negative }

    var steps = 0
    var number = number
    while number > 1 {
      number =
        number.isMultiple(of: 2)
        ? number / 2
        : number * 3 + 1
      steps += 1
    }
    return steps
  }
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
	if n < 1 {
		return 0, errors.New("invalid n")
	}
	c := 0
	for ; n > 1; c++ {
		if n%2 == 0 {
			n /= 2
		} else {
			n = 3*n + 1
		}
	}
	return c, nil
}
